model/src/main/scala/za/co/absa/atum/model/dto/traits/CheckpointCore.scala

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.atum.model.dto.traits
```

Comment:
hmm I'm not sure I like the location of this file; it has only this 1 file which doesn't really do much to our service in terms of better 'modularity' / 'structure of the DTO module. Besides, the fact that it's trait can be seen pretty nicely in IntelliJ, as it's visually different (green trait icon)

Reply:
In fact, I don't really see a big value in doing this at all to be fair. I mean, what if CheckpointV3 will be different? Then we can't do this or will update the trait..I mean it seems potentially limiting without too much benefits. 

Also you did it only for checkpoints, not Partitioning DTOs; to me it really seem unncessary

Reply:
The trait is a solution for inability/discouragement of `case class` inheritance. Because `CheckpointWithPartitioningDTO` is by definition an enhanced `CheckpointDTO`. And there's a a point, where I want to be able to handle both with the same function.

Reply:
Package `traits` is debatable of course. Naming... 😉